app/controllers/concerns/filter_parameters.rb

```diff
@@ -25,7 +19,33 @@ def deserialize_array_filter_param(param)
     params[param].split(",") if params.key?(param)
   end
 
+  def remove_previous_parameters(all_parameters)
+    all_parameters.except(*previous_parameters.keys)
```

Comment:
Minor nit: is it worth doing this?

```suggestion
    all_parameters.reject { |k,v| k.start_with? "prev_" }
```

It's more performant (fwiw here) and less cognitive load (reader doesn't need to think about `previous_parameters`).